File: Yandex/Practicum/Algorightms_and_Data_Structures/8_sprint/K._Сравнить_две_строки/solution.py
import sys


def solution() -> None:
    """
    Time Complexity O(N + M)
    Memory Complexity O(N + M)
    """
    a: str = sys.stdin.readline().rstrip()
    b: str = sys.stdin.readline().rstrip()
    a_even: list[str] = []
    b_even: list[str] = []
    for letter in a:
        if (ord(letter) - 97) & 1: a_even.append(letter)
    for letter in b:
        if (ord(letter) - 97) & 1: b_even.append(letter)
    if a_even == b_even: sys.stdout.write('0\n')
    else: sys.stdout.write(['-1', '1'][a_even > b_even] + '\n')


if __name__ == '__main__':
    solution()

File: Yandex/Practicum/Algorightms_and_Data_Structures/8_sprint/K._Сравнить_две_строки/solution.cpp
#include <iostream>
#include <string>


void solution() {
    /*
    Time Complexity O(N + M)
    Memory Complexity O(N + M)
    */
    std::string a, b;
    std::cin >> a >> b;
    std::string aEven = "";
    std::string bEven = "";
    for (char& letter : a) {
        if ((letter - 'a') & 1) aEven.push_back(letter);
    }
    for (char& letter : b) {
        if ((letter - 'a') & 1) bEven.push_back(letter);
    }
    if (aEven == bEven) std::printf("0\n");
    else std::cout << (aEven < bEven? "-1" : "1") << "\n";

}


int main() {
    solution();
}